qa/rpc-tests/sc_rpc_cmds_fee_handling.py

```diff
@@ -0,0 +1,293 @@
+#!/usr/bin/env python2
+# Copyright (c) 2014 The Bitcoin Core developers
+# Copyright (c) 2018 The Zencash developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.test_framework import MINIMAL_SC_HEIGHT, MINER_REWARD_POST_H200
+from test_framework.authproxy import JSONRPCException
+from test_framework.util import assert_true, assert_equal, initialize_chain_clean, \
+    start_nodes, sync_blocks, sync_mempools, connect_nodes_bi, mark_logs, \
+    get_epoch_data, swap_bytes 
+from test_framework.mc_test.mc_test import *
+import os
+import pprint
+from decimal import Decimal
+from test_framework.mininode import COIN
+import json
+
+NUMB_OF_NODES = 3
+DEBUG_MODE = 1
+EPOCH_LENGTH = 10
+FT_SC_FEE = Decimal('0')
+MBTR_SC_FEE = Decimal('0')
+CERT_FEE = Decimal("0.00025")
+CUSTOM_FEE_RATE_ZAT_PER_BYTE = Decimal('2.0')
+CUSTOM_FEE_RATE_ZEN_PER_KBYTE = CUSTOM_FEE_RATE_ZAT_PER_BYTE/COIN*1000
+
+class ScRpcCmdsFeeHandling(BitcoinTestFramework):
+    alert_filename = None
+
+    def setup_chain(self, split=False):
+        print("Initializing test directory " + self.options.tmpdir)
+        initialize_chain_clean(self.options.tmpdir, NUMB_OF_NODES)
+        self.alert_filename = os.path.join(self.options.tmpdir, "alert.txt")
+        with open(self.alert_filename, 'w'):
+            pass  # Just open then close to create zero-length file
+
+    def setup_network(self, split=False):
+
+        ed = "-exportdir=" + self.options.tmpdir
+        self.nodes = start_nodes(NUMB_OF_NODES, self.options.tmpdir,
+            extra_args=[
+                ['-logtimemicros=1', '-debug=sc', '-debug=py', '-debug=mempool'],
+                ['-logtimemicros=1', '-debug=sc', '-debug=py', '-debug=mempool',
+                    '-paytxfee='+str(CUSTOM_FEE_RATE_ZEN_PER_KBYTE)],
+                ['-logtimemicros=1', '-debug=sc', '-debug=py', '-debug=mempool']
+                ])
+
+        connect_nodes_bi(self.nodes, 0, 1)
+        connect_nodes_bi(self.nodes, 1, 2)
+        self.is_network_split = split
+        self.sync_all()
+
+    def run_test(self):
+        '''
+        Test the automatic fee computation for SC related transactions and certificates
+        '''
+
+        def get_fee_rate(size, fee):
+            return ((fee*COIN)/size)
+        
+        def isclose(d1, d2, tolerance=Decimal('0.005')):
+            dec1 = Decimal(d1)
+            dec2 = Decimal(d2)
+            return abs(d1-d2) <= tolerance
```

Comment:
In function abs used non-converted values of d1 and d2 instead of dec1 and dec2.

Reply:
done